src/charm.py

```diff
@@ -101,35 +113,17 @@ def _on_start(self, event):
             return
 
         if "replset_initialized" not in self._app_data:
```

Comment:
To avoid heavy nesting I'd recommend 
```suggestion
        # skip initialisation if already initialised 
        if "replset_initialized" in self._app_data:
               return 
```

Reply:
Assigned to Mykola

Reply:
fixed in https://github.com/delgod/mongodb-k8s-operator/pull/1/commits/37a9a765a1abb8048348e8bc331a2400eb615a89